eth/eth_test/common_test.go

```diff
@@ -0,0 +1,178 @@
+package eth_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/bloxapp/ssv/eth/eventsyncer"
+	"github.com/bloxapp/ssv/eth/executionclient"
+	"github.com/bloxapp/ssv/eth/simulator"
+	"github.com/bloxapp/ssv/eth/simulator/simcontract"
+	"github.com/bloxapp/ssv/monitoring/metricsreporter"
+	"github.com/bloxapp/ssv/operator/storage"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rpc"
+	"go.uber.org/zap/zaptest"
+	"math/big"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+type CommonTestInput struct {
+	t             *testing.T
+	sim           *simulator.SimulatedBackend
+	boundContract *simcontract.Simcontract
+	blockNum      *uint64
+	nodeStorage   storage.Storage
+	doInOneBlock  bool
+}
+
+func NewCommonTestInput(
+	t *testing.T,
+	sim *simulator.SimulatedBackend,
+	boundContract *simcontract.Simcontract,
+	blockNum *uint64,
+	nodeStorage storage.Storage,
+	doInOneBlock bool,
+) *CommonTestInput {
+	return &CommonTestInput{
+		t:             t,
+		sim:           sim,
+		boundContract: boundContract,
+		blockNum:      blockNum,
+		nodeStorage:   nodeStorage,
+		doInOneBlock:  doInOneBlock,
+	}
+}
+
+type testEnv struct {
+	eventSyncer   *eventsyncer.EventSyncer
+	validators    []*testValidatorData
+	ops           []*testOperator
+	nodeStorage   storage.Storage
+	sim           *simulator.SimulatedBackend
+	boundContract *simcontract.Simcontract
+	auth          *bind.TransactOpts
+	shares        [][]byte
+	execClient    *executionclient.ExecutionClient
+	rpcServer     *rpc.Server
+	httpSrv       *httptest.Server
+}
+
+func setupEnv(
+	t *testing.T,
+	ctx context.Context,
+	testAddresses []*ethcommon.Address,
+	validatorsCount uint64,
+	operatorsCount uint64,
+) (*testEnv, error) {
+	logger := zaptest.NewLogger(t)
+
+	// Create operators RSA keys
+	ops, err := createOperators(operatorsCount, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	validators := make([]*testValidatorData, validatorsCount)
+	shares := make([][]byte, 10)
```

Comment:
Does the exact value mean anything or is it just an arbitrary number?

Reply:
Correct. I've removed this. Using the `validatorsCount` and `operatorsCount` params instead

Reply:
Fixed